roles/uperf/tasks/start_client.yml

```diff
@@ -21,8 +26,16 @@
     vars: 
       resource_item: "{{ server_pods.resources }}"
     when: 
-      - ( workload_args.serviceip|default(False) == False and server_pods.resources|length > 0 ) or 
-        ( workload_args.serviceip|default(False) == True and server_pods.resources|length > 0 and 
+      - ( workload_args.serviceip|default(False) == False and server_pods.resources|length > 0 )
+
+
+  - name: Start Client(s) with nodeport serviceIP
+    k8s:
+      definition: "{{ lookup('template', 'workload.yml.j2') | from_yaml }}"
+    vars:
+      resource_item: "{{ server_pods.resources }}"
+    when:
+      - ( workload_args.serviceip|default(False) == True and server_pods.resources|length > 0 and 
```

Comment:
Can we combine this and the above task?

Reply:
While it makes sense from a code perspective, I intentionally split it out since the uperf code is complex enough. Didn't want to combine a service related task with a pod related task.